certification-content-bundle-spec.md

```diff
@@ -0,0 +1,118 @@
+audience           |          | dictionary | A locale dictionary of audiences
+prerequisites      |          | dictionary | A locale dictionary of prerequisites
+tags               |          | array      | Array of strings to be used as hints in searching, etc.
+product_tags       |          | array      | Array of strings from the "Products" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+role_tags          |          | array      | Array of strings from the "Roles" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+domain_tags        |          | array      | Array of strings from the "Domain" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+credits            |          | integer    | Price of the certification
+certificate_award  | ✓        | string     | A unique identifier for the corresponding award that this certification grants
+steps              | ✓        | array      | An array of `steps` (see [Steps](#steps) below for details)
+
+### Steps
+
+A certification is composed of multiple `steps`. A step can be either a Course Template or an exam that the user must complete to progress through the certification.
+
+#### Gated Steps
+
+Steps can optionally be marked as "gated". A gated step and all of its succeeding steps will be unavailable to the user until all preceding steps are completed.
+
+For example, consider the following order of steps, where steps `C` and `E` are gated:
+
+```
+A -- B -- |C| -- D -- |E|
+```
+
+- At first, only `A` and `B` are available to the user.
+- Once `A` and `B` are completed (in any order), `C` and `D` become available.
+- Once `C` and `D` are completed (in any order), `E` becomes available.
+
+#### Course Step Example
+
+```yml
+id: step-0
+type: course_template
+content_id: qwiklabs-test-content/my-cool-course
+```
+
+#### Exam Step Example
+
+```yml
+id: step-1
+gated: true
+type: course_template
+content_id: qwiklabs-test-content/my-excellent-exam
+proctor: qwiklabs-live-plus
+```
+
+attribute          | required | type       | notes
+-------------------| -------- | ---------- | -----------------------------------------
+id                 | ✓        | string     | A unique identifier for this step, to be consistent across revisions
+gated              |          | boolean    | Whether this step should be "gated" (see [Gated Steps](#gated-steps) above for details); `false` by default
+type               | ✓        | enum       | One of ["course_template", "exam"]
+content_id         | ✓        | string     | Unique identifier of this step's corresponding CourseTemplate or Exam
+proctor            |          | enum       | (exam steps only) Which service to use for proctoring this step's corresponding Exam. One of ["qwiklabs-live-plus", "qwiklabs-record-plus"]
```

Comment:
Is this an authoring concern?

Reply:
Yes, the level of proctoring is determined by authors depending on the needs of their certification.

Reply:
The "level" of proctoring makes sense to me as an authoring concern, but the specific "proctoring service" doesn't

Reply:
We don't expect authors to be defining this, really to define the offering.

Reply:
Yeah, I guess I didn't expect us to be using content bundles for offerings. Probably something that warrants further discussion.